tests/integration/security/sds_tls_origination/egress_gateway_origination_test.go

```diff
@@ -0,0 +1,286 @@
+				ClientCert: sdstlsutil.MustReadCert(t, "cert-chain.pem"),
+				PrivateKey: sdstlsutil.MustReadCert(t, "key.pem"),
+				CaCert:     sdstlsutil.MustReadCert(t, "root-cert.pem"),
+			}
+			// Configured with an invalid ClientCert
+			var credentialBCert = sdstlsutil.TLSCredential{
+				ClientCert: sdstlsutil.FakeCert,
+				PrivateKey: sdstlsutil.MustReadCert(t, "key.pem"),
+				CaCert:     sdstlsutil.MustReadCert(t, "root-cert.pem"),
+			}
+			// Configured with an invalid ClientCert and PrivateKey
+			var credentialBCertAndKey = sdstlsutil.TLSCredential{
+				ClientCert: sdstlsutil.FakeCert,
+				PrivateKey: sdstlsutil.FakeKey,
+				CaCert:     sdstlsutil.MustReadCert(t, "root-cert.pem"),
+			}
+			// Add kubernetes secret to provision key/cert for gateway.
+			sdstlsutil.CreateKubeSecret(t, ctx, []string{credNameGeneric}, "MUTUAL", credentialAGeneric, false)
+			defer ingressutil.DeleteKubeSecret(t, ctx, []string{credNameGeneric})
+
+			sdstlsutil.CreateKubeSecret(t, ctx, []string{credNameNotGeneric}, "MUTUAL", credentialANonGeneric, true)
+			defer ingressutil.DeleteKubeSecret(t, ctx, []string{credNameNotGeneric})
+
+			sdstlsutil.CreateKubeSecret(t, ctx, []string{fakeCredNameA}, "MUTUAL", credentialBCert, false)
+			defer ingressutil.DeleteKubeSecret(t, ctx, []string{fakeCredNameA})
+
+			sdstlsutil.CreateKubeSecret(t, ctx, []string{fakeCredNameB}, "MUTUAL", credentialBCertAndKey, false)
+			defer ingressutil.DeleteKubeSecret(t, ctx, []string{fakeCredNameB})
+
+			sdstlsutil.CreateKubeSecret(t, ctx, []string{simpleCredName}, "SIMPLE", credentialASimple, false)
+			defer ingressutil.DeleteKubeSecret(t, ctx, []string{simpleCredName})
+
+			internalClient, externalServer, _, serverNamespace := sdstlsutil.SetupEcho(t, ctx)
+
+			// Set up Host Namespace
+			host := "server." + serverNamespace.Name() + ".svc.cluster.local"
+
+			testCases := map[string]struct {
+				response        []string
+				credentialToUse string
+				gateway         bool // true if the request is expected to be routed through gateway
+			}{
+				// Use CA certificate and client certs stored as k8s secret with the same issuing CA as server's CA.
+				// This root certificate can validate the server cert presented by the echoboot server instance and server CA can
+				// validate the client cert. Secret is of type generic.
+				"MUTUAL TLS with correct root cert and client certs and generic secret type": {
+					response:        []string{response.StatusCodeOK},
+					credentialToUse: strings.TrimSuffix(credNameGeneric, "-cacert"),
+					gateway:         true,
+				},
+				// Use CA certificate and client certs stored as k8s secret with the same issuing CA as server's CA.
+				// This root certificate can validate the server cert presented by the echoboot server instance and server CA can
+				// validate the client cert. Secret is of type generic.
+				"MUTUAL TLS with correct root cert and client certs and non generic secret type": {
```

Comment:
The case name is non generic secret type, but the comment mentioned that secret is generic.

Reply:
Good catch! I'll push a fix